recipes/pdal/all/conanfile.py

```diff
@@ -33,248 +36,322 @@ class PdalConan(ConanFile):
         "fPIC": True,
         "with_unwind": False,
         "with_xml": True,
-        "with_lazperf": False, # TODO: should be True
-        "with_laszip": True,
         "with_zlib": True,
         "with_lzma": False,
         "with_zstd": True,
     }
 
-    generators = "cmake", "cmake_find_package"
-
     @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def _min_cppstd(self):
+        return 17
 
     @property
-    def _is_msvc(self):
-        return str(self.settings.compiler) in ["Visual Studio", "msvc"]
+    def _compilers_minimum_version(self):
+        return {
+            "Visual Studio": "16",
+            "msvc": "192",
+            "gcc": "8",
+            "clang": "7",
+            "apple-clang": "12.0",
+        }
 
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        copy(self, "CMakeLists.txt", src=self.recipe_folder, dst=self.export_sources_folder)
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
         if self.settings.os not in ["Linux", "FreeBSD"]:
-            del self.options.with_unwind
+            self.options.rm_safe("with_unwind")
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        # TODO package improvements:
-        # - switch from vendored arbiter (not in CCI). disabled openssl and curl are deps of arbiter
-        # - switch from vendor/nlohmann to nlohmann_json (in CCI)
-        self.requires("boost/1.78.0")
-        self.requires("eigen/3.4.0")
-        self.requires("gdal/3.4.1")
-        self.requires("libcurl/7.80.0") # mandatory dependency of arbiter (to remove if arbiter is unvendored)
+        self.requires("arbiter/cci.20231005", transitive_headers=True, transitive_libs=True)
```

Comment:
It would be great to update the requirements list to this:
```
diff --git a/recipes/pdal/all/conanfile.py b/recipes/pdal/all/conanfile.py
index 9f2effda1..1762ed4bf 100644
--- a/recipes/pdal/all/conanfile.py
+++ b/recipes/pdal/all/conanfile.py
@@ -73,15 +73,15 @@ class PdalConan(ConanFile):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("arbiter/cci.20231005", transitive_headers=True, transitive_libs=True)
+        self.requires("arbiter/cci.20231122", transitive_headers=True, transitive_libs=True)
         self.requires("boost/1.83.0")
         self.requires("eigen/3.4.0", transitive_headers=True, transitive_libs=True)
-        self.requires("gdal/3.8.0", transitive_headers=True, transitive_libs=True)
-        self.requires("json-schema-validator/2.2.0")
+        self.requires("gdal/3.8.3", transitive_headers=True, transitive_libs=True)
+        self.requires("json-schema-validator/2.3.0")
         self.requires("libgeotiff/1.7.1")
         self.requires("nanoflann/1.5.0", transitive_headers=True, transitive_libs=True)
-        self.requires("nlohmann_json/3.11.2", transitive_headers=True, transitive_libs=True)
-        self.requires("proj/9.3.0", transitive_headers=True, transitive_libs=True)
+        self.requires("nlohmann_json/3.11.3", transitive_headers=True, transitive_libs=True)
+        self.requires("proj/9.3.1", transitive_headers=True, transitive_libs=True)
         self.requires("utfcpp/4.0.1")
         if self.options.with_xml:
             self.requires("libxml2/2.11.5", transitive_headers=True, transitive_libs=True)
```

this is based on https://github.com/conan-io/conan-center-index/pull/21271 and https://github.com/conan-io/conan-center-index/pull/19298